src/components/validation/runner/mod.rs

```diff
@@ -532,7 +562,35 @@ fn spawn_input_driver(
 
             // Update the runner metrics for the sent event. This will later
             // be used in the Validators, as the "expected" case.
-            let mut input_runner_metrics = input_runner_metrics.lock().unwrap();
+            let mut input_runner_metrics = input_runner_metrics.lock().await;
+
+            let (modified, mut event) = match input_event.clone() {
+                TestEvent::Passthrough(event) => (false, event),
+                TestEvent::Modified { modified, event } => (modified, event),
+            };
+
+            event
+                .as_log()
+                .get_timestamp()
+                .unwrap()
+                .as_timestamp_unwrap();
+
+            // the controlled edge (vector source) adds metadata to the event when it is received.
+            // thus we need to add it here so the expected values for the comparisons on transforms
+            // and sinks are accurate.
+            if source_is_controlled_edge {
+                if let Event::Log(ref mut log) = event {
+                    log_namespace.insert_standard_vector_source_metadata(log, "vector", now);
+                }
+            }
+
+            let input_event = match &input_event {
+                TestEvent::Passthrough(_) => TestEvent::Passthrough(event.clone()),
+                TestEvent::Modified { modified, event: _ } => TestEvent::Modified {
+                    modified: *modified,
+                    event: event.clone(),
+                },
+            };
```

Comment:
Could you potentially just move this logic up to the first match block? You'd be able to save a lot of duplicate code, and a clone of the event, by doing so.

Reply:
👍  I cleaned this up a bit.